staging/src/k8s.io/apiserver/pkg/cel/library/image.go

```diff
@@ -0,0 +1,312 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"fmt"
+
+	"github.com/distribution/reference"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+const (
+	// DefaultTag is the tag name that will be used if no tag provided
+	DefaultTag = "latest"
+)
+
+// Image provides a CEL function library extension for parsing image references.
+//
+// image
+//
+// Converts a string to an image reference or returns an error if the image is not a valid image reference. Refer
+// to github.com/opencontainers/image-spec documentation for information on accepted image references.
+// An optional "normalize" argument can be passed to enable normalization. Normalization parses the image
+// into a normalized image. An example normalized image is "docker.io/library/ubuntu" for image "ubuntu".
+//
+//	image(<string>) <Image>
+//
+// Examples:
+//
+//	image('registry.k8s.io/kube-apiserver-arm64:latest') // Returns an Image
+//	image('nginx') // Returns an Image
+//	image('registry.k8s.io/kube-apiserver-arm64@sha256:6aefddb645ee6963afd681b1845c661d0ea4c3b20ab9db86d9e753b203d385f2') // Returns an Image
+//	image('registry.k8s.io/kube-apiserver-arm64:latest@sha256:6aefddb645ee6963afd681b1845c661d0ea4c3b20ab9db86d9e753b203d385f2') // Returns an Image
+//	image('registry.k8s.io//kube-apiserver-arm64') // error
+//	image(0) // error
+///	image('ubuntu', true) // Applies normalization and returns components of "docker.io/library/ubuntu".
+///	image('ubuntu', false) // Returns Image for "ubuntu".
```

Comment:
worth mentioning that image('ubuntu') defaults to no normalization behavior which returns Image?